transport/tchannel/handler.go

```diff
@@ -213,7 +221,7 @@ func (rw *responseWriter) AddHeaders(h transport.Headers) {
 }
 
 func (rw *responseWriter) addHeader(key string, value string) {
-	rw.headers = rw.headers.With(key, value)
+	rw.headers[key] = value
```

Comment:
why this change?

Reply:
because how rw.headers becomes a `map[string]string` and it will hold the case sensitive header as keys. It was a Headers object.